splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -351,6 +351,25 @@ public void createKeyColumnUseViewInSysIBM(TransactionController tc) throws Stan
         SpliceLogUtils.info(LOG, "View SYSKEYCOLUSE in SYSIBM is created!");
     }
 
+    public void createIndexColumnUseViewInSysCat(TransactionController tc) throws StandardException {
+        TableDescriptor td = getTableDescriptor("INDEXCOLUSE", sysCatSchemaDesc, tc);
+
+        // drop it if it exists
+        if (td != null) {
+            ViewDescriptor vd = getViewDescriptor(td);
+
+            // drop the view deifnition
+            dropAllColumnDescriptors(td.getUUID(), tc);
+            dropViewDescriptor(vd, tc);
+            dropTableDescriptor(td, sysCatSchemaDesc, tc);
+        }
+
```

Comment:
If it already exists, skip dropping and recreating the view.

Reply:
I accidentally approved it. Changing to "request cahnges"

Reply:
Hi Jun, what do you mean by "recreating the view"? I mean, if the existing one is not dropped, we cannot create the view with the same name (`createOneSystemView` fails). Or do I miss anything here?

Reply:
Hi Jun, as we talked previously, I tested doing an upgrade from 3.0.0.1989 to 3.2.0.1989 locally. Here is what I did:

1. compile DB-10976-3.0 branch and start a cdh6.3.0 cluster
2. check SYSCAT.INDEXCOLUSE is there and query on it works (108 rows in fresh start state)
3. copy platform_it/target to another place
4. compile DB-10976 branch and start a cdh6.3.0 cluster, verify its version and the existence of SYSCAT_INDEXCOLUSE
5. shutdown the cluster and replace platform_it/target with the previously copied 3.0.0.1989 version
6. start cdh6.3.0 cluster again using -b on the 3.0 persistent files
7. cluster started normally and verify the existence of SYSCAT_INDEXCOLUSE , query on it should return 108 rows